lib/rubocop/cop/lint/memoized_instance_variable_name.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for memoized methods whose instance variable name
+      # does not match the method name.
+      #
+      # @example
+      #
```

Comment:
Check out the yard documentation format here - https://github.com/bbatsov/rubocop/blob/master/manual/development.md#documentation .  You should remove the extra line after `@example`.  And since there is only one style here, you only need to specify `@example` once.

Keep the extra description, it's great!